libs/editor/egui_editor/src/events.rs

```diff
@@ -761,21 +761,20 @@ fn calc_modification(
                             {
                                 cursor.pos = click_offset;
                             }
-
                             cursor.advance_word(
-                                false,
+                                true,
                                 &buffer.current,
                                 &buffer.current.segs,
                                 galleys,
                             );
-                            let end_of_word_pos = cursor.pos;
+                            let begin_of_word_pos = cursor.pos;
                             cursor.advance_word(
-                                true,
+                                false,
                                 &buffer.current,
                                 &buffer.current.segs,
                                 galleys,
                             );
```

Comment:
Preferring the previous word when the double click occurs on a word boundary makes more sense when clicking at the end of a line (select the last word on this line instead of the first word on the next line). Double-clicking on the first character position of a line now selects the last word on the previous line, which is fine for now